ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java

```diff
@@ -443,12 +443,12 @@ public static BlobProperties buildBlobProperties(Map<String, Object> args) throw
           // key size
           sizeToAllocate += 4;
           String keyToStore = key.substring(Headers.USER_META_DATA_HEADER_PREFIX.length());
-          sizeToAllocate += keyToStore.getBytes(StandardCharsets.US_ASCII).length;
+          sizeToAllocate += keyToStore.getBytes(StandardCharsets.UTF_8).length;
```

Comment:
I thought of one slight deployment issue with changing the charset without making a new user metadata version.

If you deploy this change to one frontend, a user could write a non-ascii value into user metadata, and the binary format will now store its real utf-8 value instead of a `?` character.
If this blob's user metadata is fetched on a frontend without this change, that frontend will attempt to decode as ascii and a deserialization error will occur.

However, this may be a negligible number of users affected and any problems will be resolved by completing the deployment, so maybe continuing with this approach and opting for a fast deployment is the easiest solution.

Reply:
Yeah, this is indeed a deployment issue as frontends can't be deployed simultaneously. The edge case only affects China team for now. I guess the impact should be very minimum.  (I agree introducing metadata v2 is theoretically the safest way to migrate from ascii to utf-8 smoothly, however we can tolerate the transient error as only one team is really building metadata with non-ascii character, we can coordinate with them when rolling out this change).